sentry_sdk/integrations/spark/spark_worker.py

```diff
@@ -58,25 +58,37 @@ def _capture_exception(exc_info, hub):
 def _tag_task_context():
     from pyspark.taskcontext import TaskContext
 
-    taskContext = TaskContext.get()
-
-    if taskContext:
-        with configure_scope() as scope:
-            scope.set_tag("stageId", taskContext.stageId())
-            scope.set_tag("partitionId", taskContext.partitionId())
-            scope.set_tag("attemptNumber", taskContext.attemptNumber())
-            scope.set_tag("taskAttemptId", taskContext.taskAttemptId())
-
-            if "app_name" in taskContext._localProperties:
-                scope.set_tag("app_name", taskContext._localProperties["app_name"])
-                scope.set_tag(
-                    "application_id", taskContext._localProperties["application_id"]
-                )
-
-            if "callSite.short" in taskContext._localProperties:
-                scope.set_extra(
-                    "callSite", taskContext._localProperties["callSite.short"]
-                )
+    with configure_scope() as scope:
+
+        @scope.add_event_processor
+        def process_event(event, hint):
```

Comment:
Same here, please just wrap the entire event processor in `capture_internal_exceptions`